index.js

```diff
@@ -1,18 +1,55 @@
-import fetch from 'node-fetch';
-// Recommend using node-fetch for those familiar with JS fetch
-
-const COLORS = 'https://nt-cdn.s3.amazonaws.com/colors.json';
-
-/**
- * @param name filter for color name
- * @param hex filter for color hex code
- * @param compName filter for complementary color name
- * @param compHex filter for complementary color hex code
- * @returns Promise
- */
-const fetchColors = ({ name, hex, compName, compHex }) => {
-  throw Error('Not implemented');
+import fetch from "node-fetch";
+
+const fetchColors = async ({ name, hex, compName, compHex }) => {
+  try {
+    const response = await fetch("https://nt-cdn.s3.amazonaws.com/colors.json");
+    const colors = await response.json();
+
+    const filteredColors = colors.filter((color) => {
+      // check the name
+      const nameMatches = name
+        ? color.name.toLowerCase() === name.toLowerCase()
```

Comment:
you can outsource this to a function. if the requirement changed to be case sensitive, you have to change it in 4 places